src/WebJobs.Extensions.DurableTask/OrchestrationServiceFactory.cs

```diff
@@ -179,12 +182,46 @@ public EmulaterOrchestrationServiceFactory(DurableTaskOptions options)
 
             public IOrchestrationServiceClient GetOrchestrationClient(OrchestrationClientAttribute attribute)
             {
-                return this.service;
+                return (IOrchestrationServiceClient)this.service;
             }
 
             public IOrchestrationService GetOrchestrationService()
             {
-                return this.service;
+                return (IOrchestrationService)this.service;
+            }
+        }
+
+        private class RedisOrchestrationServiceFactory : IOrchestrationServiceFactory
+        {
+            private readonly RedisOrchestrationService defaultTaskHubService;
+            private readonly string redisConnectionString;
+
+            public RedisOrchestrationServiceFactory(DurableTaskOptions options, IConnectionStringResolver connectionStringResolver)
+            {
+                this.redisConnectionString = connectionStringResolver.Resolve(options.StorageProvider.Redis.ConnectionStringName);
+                this.defaultTaskHubService = new RedisOrchestrationService(new RedisOrchestrationServiceSettings()
+                {
+                    TaskHubName = options.HubName,
+                    RedisConnectionString = this.redisConnectionString,
+                });
+            }
+
+            public IOrchestrationServiceClient GetOrchestrationClient(OrchestrationClientAttribute attribute)
+            {
+                if (string.IsNullOrEmpty(attribute.TaskHub))
```

Comment:
In addition to this, if a `TaskHub` value is specified and its the same name as `defaultTaskHubService.HubName`, we should also return `this.defaultTaskHubService`.

Reply:
Good catch. That may have been the reason behind one of the scenarios not working that I thought should have.